addon/pods/components/frost-tab/component.js

```diff
@@ -1,21 +1,55 @@
 import Ember from 'ember'
 import layout from './template'
-import FrostTabs from 'ember-frost-tabs/pods/components/frost-tabs/component'
+// import FrostTabs from 'ember-frost-tabs/pods/components/frost-tabs/component'
+import { PropTypes } from 'ember-prop-types'
+
+const {
+  Component
+} = Ember
+
+export default Component.extend({
+  // == Component properties ==================================================
 
-export default Ember.Component.extend({
   layout: layout,
-  classNames: ['content'],
-  classNameBindings: ['isSelected::hidden', 'tabClassNames'],
+  classNames: ['frost-tab'],
+  classNameBindings: ['tabClassNames'],
 
-  frostTabs: null,
+  // == State properties ======================================================
+
+  propTypes: {
+    id: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+    content: PropTypes.oneOfType([
+      PropTypes.EmberObject,
+      PropTypes.object
+    ]).isRequired,
+    disabled: PropTypes.bool,
+    // Set by the parent component
+    hook: PropTypes.string,
+    targetOutlet: PropTypes.string.isRequired,
+    selectedTab: PropTypes.string,
+    onChange: PropTypes.func
+  },
 
-  init () {
-    this._super(...arguments)
-    this.set('frostTabs', this.nearestOfType(FrostTabs))
-    this.get('frostTabs').register(this)
+  getDefaultProps () {
+    return {
+      disabled: false
+    }
   },
 
-  isSelected: Ember.computed('frostTabs.selection', function () {
-    return this.get('id') === this.get('frostTabs.selection')
-  })
+  // == Computed properties ===================================================
+
+  isSelected: Ember.computed('id', 'selectedTab', function () {
+    return this.id === this.selectedTab && !this.disabled
+  }),
+
+  // == Actions ===============================================================
+
+  actions: {
+    change () {
+      if (this.onChange) {
```

Comment:
I was thinking that checking the type will be duplication of the effort since we are already doing this with propTypes.

I did try to put the following in the template and it was not working:
```
onClick=onChange
```
or
```
onClick=(action onChange)
```